### PR TITLE
fix(types): abci valset update abci does not need a public key on replay

### PR DESCRIPTION
**Pull request by lklimek**
## Issue being fixed or feature implemented

Tenderdash requires validator set updates to contain public keys of all validators. 
However, this should not be required, as in some scenarios (like replay) these
pubkeys are not available.

## What was done?

Validators in validator set no longer require public key.

## How Has This Been Tested?

On testnet, when got the failure, with the following procedure:

```bash
dashmate config set --config testnet platform.drive.tenderdash.docker.image dashpay/tenderdash:fix-validator-update-without-pubkey
dashmate stop --platform
dashmate reset --platform
dashmate start --platform
```

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: types/validator_set_test.go [hunk 1]
```diff
@@ -161,7 +161,7 @@ func TestValidatorSetValidateBasic(t *testing.T) {
 				Proposer:           badValNoPublicKey,
 			},
 			err: true,
-			msg: "invalid validator pub key #0: validator does not have a public key",
+			msg: "thresholdPublicKey error: validator public key is not set",
```
Comment (shumkov) [inline types/validator_set_test.go:164]: Why do we expect an error that thresholdPublicKey is missing when we do not pass validator private key?

File: types/validator.go [hunk 2]
```diff
@@ -136,12 +136,9 @@ func (v *Validator) ValidateBasic() error {
...
 
 // ValidatePubKey performs basic validation on the public key.
+// Note that it is not an error for the public key to be nil.
 func (v *Validator) ValidatePubKey() error {
-	if v.PubKey == nil {
-		return errors.New("validator does not have a public key")
-	}
-
-	if len(v.PubKey.Bytes()) != bls12381.PubKeySize {
+	if v.PubKey != nil && len(v.PubKey.Bytes()) != bls12381.PubKeySize {
```
Comment (shumkov) [inline types/validator.go:141]: Why we combine these two errors?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: types/validator.go (hunk 2 above)
Comment (lklimek) [reply types/validator.go]: changed the solution, so this comment is outdated

File: types/protobuf.go [hunk 3]
```diff
@@ -169,7 +169,14 @@ func (pb2tm) ValidatorSetFromProtoUpdate(
 			hasPublicKeys = false
 			break
 		}
+
+		pubkey, err := cryptoenc.PubKeyFromProto(*v.PubKey)
+		if err != nil || len(pubkey.Bytes()) == 0 {
```
Comment (shumkov) [inline types/protobuf.go:174]: I wouldn't hide error if key is not valid

File: types/validator_set.go [hunk 4]
```diff
@@ -199,6 +199,9 @@ func (vals *ValidatorSet) ThresholdPublicKeyValid() error {
 		return errors.New("threshold public key is wrong size")
 	}
 	if len(vals.Validators) == 1 && vals.HasPublicKeys {
+		if vals.Validators[0].PubKey == nil {
```
Comment (shumkov) [inline types/validator_set.go:202]: Why do we validate it only in case of only 1 node?

Comment (shumkov) [reply types/validator_set.go:202]: My bad. Understood now.

File: types/protobuf.go (hunk 3 above)
Comment (lklimek) [reply types/protobuf.go]: done

**Review by shumkov — APPROVED** [review]

**Pull request merged**